api/src/main/java/org/apache/iceberg/expressions/Literal.java

```diff
@@ -71,6 +72,10 @@ static Literal<BigDecimal> of(BigDecimal value) {
     return new Literals.DecimalLiteral(value);
   }
 
+  static Literal<VariantData> of(VariantData value) {
```

Comment:
You should use Variant interface instead of VariantData. So you don't need to expose VariantData as public class.